src/main/java/com/ea/restaurant/service/impl/MongoOrderStatusHistoryServiceImpl.java

```diff
@@ -32,4 +32,10 @@ public void updateBatchToProcessed(Set<ObjectId> ids) {
     }
     this.mongoOrderStatusHistoryRepository.saveAll(mongoOrderStatusHistories);
   }
+
+  @Override
+  public void insertMongoOrderStatusHistories(
```

Comment:
I think this should be transactional including a rollback for.

Also this should be made in another pr not in this one

Reply:
Done.